backend/entityservice/api_def/swagger.yaml

```diff
@@ -509,25 +509,26 @@ paths:
 
         ### result_type = "similarity_scores"
 
-        The list of the indices of potential matches and their similarity score
+        The list of the candidate of potential matches and their similarity score
         where the similarity score is greater than the mapping threshold.
         Data is returned as `json` object e.g.,
 
             {
                 "similarity_scores":
                     [
-                        [5, 27, 1.0],
-                        [14, 10, 1.0]
+                        {'group': [[0, 5], [1, 27]], 'sim': 1.0},
+                        {'group': [[1, 10], [0, 14]], 'sim': 1.0}
```

Comment:
This format has a lot of overhead, as it is writing 'group' and 'sim' for every pair.
How about 
```
{
  "similarity_scores":
  [
    [[0, 5], [1, 27], 1.0], 
    [[1, 10], [0, 14], 1.0]
  ]
}

also, I don't think 'group' is a good choice, as when we extend to the multiparty case, this gets really confusing.
The groups result shows actual groups of 2 or more entities, whereas this will only ever show pairwise results.

Reply:
I didn't realise it was valid JSON to have different value types in a single array, but it reduces the size of the output drastically, and simplify the discussion about the keys to use: none :)